File: backend/src/main/kotlin/ar/huerta/services/UserDetailsServiceImpl.java
package ar.huerta.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class UserDetailsServiceImpl implements UserDetailsService {

    @Override
    @Transactional
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        /* Administrator user = userRepository.findByUsernameAndActive(username, true);
        if (user == null) {
            throw new UsernameNotFoundException(username);
        }

        List<SimpleGrantedAuthority> roles = user.getRoles().stream().map(r -> new SimpleGrantedAuthority(r.getRoleName()))
                .collect(Collectors.toList());
        return new User(user.getUsername(), user.getPassword(), roles);

         */
                return null;
    }
}

File: backend/src/main/kotlin/ar/huerta/services/UserDetailsServiceImpl.kt
package ar.huerta.services

import ar.huerta.repositories.UserRepository
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.User
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service
import javax.transaction.Transactional

@Service
class UserDetailsServiceImpl : UserDetailsService {

    @Autowired
    private lateinit var userRepository: UserRepository;

    private val LOGGER: Logger = LoggerFactory.getLogger(UserDetailsServiceImpl::class.java)


    @Transactional
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails {
        val user: ar.huerta.entities.User?;
        try {
            user = userRepository.findByUsernameAndActive(username, true)
            val roles = user?.roles?.map { it -> SimpleGrantedAuthority(it.roleName) };
            return User(user?.username, user?.password, roles)
        } catch (e: Exception) {
            LOGGER.error("Ocurrio un error al recuperar el usuario de la base de datos", e);
            throw e;
        }

    }
}